File: notification/domain/notification.go
package domain

import "time"

// ==================== 通知模板 ====================

type NotificationTemplate struct {
	ID       int64
	TenantID int64  // 0=平台模板
	Code     string // 模板编码：welcome_sms, order_paid_merchant 等
	Channel  int32  // 1-短信 2-邮件 3-站内信
	Title    string
	Content  string // 模板内容，支持 Go text/template 占位符
	Status   int32  // 1-启用 2-停用
	Ctime    time.Time
	Utime    time.Time
}

// ==================== 通知记录 ====================

type Notification struct {
	ID       int64
	UserID   int64
	TenantID int64
	Channel  int32 // 1-短信 2-邮件 3-站内信
	Title    string
	Content  string
	IsRead   bool
	Status   int32 // 1-待发送 2-已发送 3-发送失败
	Ctime    time.Time
	Utime    time.Time
}

File: admin-frontend/src/api/notification.ts
import { request } from './client'
import type { NotificationTemplate } from '@/types/notification'

export interface ListTemplatesParams {
  tenantId?: number
  channel?: number
}

export async function listTemplates(params: ListTemplatesParams) {
  return request<NotificationTemplate[]>({
    method: 'GET',
    url: '/notification-templates',
    params: { tenant_id: params.tenantId, channel: params.channel },
  })
}

export interface TemplateData {
  tenantId?: number
  code: string
  channel: number
  title: string
  content: string
  status?: number
}

export async function createTemplate(data: TemplateData) {
  return request<{ id: number }>({
    method: 'POST',
    url: '/notification-templates',
    data: { tenant_id: data.tenantId, code: data.code, channel: data.channel, title: data.title, content: data.content, status: data.status },
  })
}

export async function updateTemplate(id: number, data: TemplateData) {
  return request<null>({
    method: 'PUT',
    url: `/notification-templates/${id}`,
    data: { tenant_id: data.tenantId, code: data.code, channel: data.channel, title: data.title, content: data.content, status: data.status },
  })
}

export async function deleteTemplate(id: number) {
  return request<null>({
    method: 'DELETE',
    url: `/notification-templates/${id}`,
  })
}

export interface SendNotificationData {
  userId: number
  tenantId?: number
  templateCode: string
  channel: number
  params?: Record<string, string>
}

export async function sendNotification(data: SendNotificationData) {
  return request<{ id: number }>({
    method: 'POST',
    url: '/notifications/send',
    data: { user_id: data.userId, tenant_id: data.tenantId, template_code: data.templateCode, channel: data.channel, params: data.params },
  })
}
